### PR TITLE
fish support

### PR DESCRIPTION
**Pull request by xu-cheng**
Another take of #15

cc @bfontaine @lordlycastle

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,17 @@
-language: bash
-before_script:
-  - "sudo apt-get install zsh"
+language: objective-c # force OS X
+before_install:
+  - brew update
+  - brew config
+install:
+  - mkdir -p $(brew --repo)/Library/Taps/homebrew
+  - ln -s $PWD $(brew --repo)/Library/Taps/homebrew/homebrew-command-not-found
```
Comment (bfontaine) [inline .travis.yml]: Could we do `brew tap homebrew/homebrew-command-not-found $PWD`?

Comment (xu-cheng) [inline .travis.yml]: No, we can't. We need to test current version instead of master version.

FYI, I haven't get test work yet.

**Comment by xu-cheng** [comment]
Stange, I finally make fish test workable. But I now have trouble to make bash passing the test. Any idea on what's the cause?

File: README.md [hunk 2]
```diff
@@ -24,12 +24,21 @@ The program 'when' is currently not installed. You can install it by typing:
...
 
 First, tap this repository:
+```bash
+brew tap homebrew/command-not-found
+```
 
-    brew tap homebrew/command-not-found
-
-Then add following line to your `.bashrc`(bash) or `.zshrc`(zsh):
+* **Bash and Zsh**: Add following line to your `.bashrc`(bash) or `.zshrc`(zsh):
```
Comment (bfontaine) [inline README.md]: > Add **the** following line

Comment (xu-cheng) [inline README.md]: Fixed

File: handler.fish [hunk 3]
```diff
@@ -0,0 +1,21 @@
+function __fish_command_not_found_on_interactive --on-event fish_prompt
+    functions --erase __fish_command_not_found_handler
```
Comment (lordlycastle) [inline handler.fish]: Some people have custom command not found handler function, so I don't think it's best to erase it. Perhaps you could call the function something different, it should work as it's triggered by the `fish_command_not_found` event and not by name.

Comment (xu-cheng) [inline handler.fish]: This is used to erase the default `__fish_command_not_found_handler` so we won't have duplicated `command not found` message.

> Some people have custom command not found handler function

Can you show me an example about that? If they need a hook, shouldn't them just use `--on-event fish_command_not_found`? I don't think people want to have multiple command not found functions at the same time.

Comment (bfontaine) [inline handler.fish]: In the Bash/Zsh script we override the command-not-found handler if there’s already one.

**Comment by jiegec** [comment]
One month. Any progress?

**Comment by xu-cheng** [comment]
cc @bfontaine Any thought?

**Comment by bfontaine** [comment]
Sorry for the delay; thank you both! :tada:

**Comment by jiegec** [comment]
Great. Working. Thank for your work!

**Pull request closed without merging**